### PR TITLE
Canada Post Address Finder #4153

### PR DESCRIPTION
**Pull request by nitheesh-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/4153

*Description of changes:*
- Canada Post Address Finder#4153
- used BaseAddress component from sbc-common-components
- bugfix: Redirecting all users to Create Account if the user nerither has account nor an active status, need to send bceid user to setup-non-bcsc-account page

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/public/index.html [hunk 1]
```diff
@@ -16,4 +18,8 @@
     <div id="app"></div>
     <!-- built files will be auto injected -->
   </body>
+  <script type="text/javascript">
```
Comment (sumesh-aot) [inline auth-web/public/index.html:21]: Is this correct, adding address complete key in source code ?

**Comment by sumesh-aot** [comment]
Can we start adding unit test cases ? I believe entities should be having some tests in place.

File: auth-web/src/services/user.services.ts [hunk 2]
```diff
@@ -90,11 +91,12 @@ export default class UserService {
   }
 
   static async createNotaryDetails (documentId: String, notaryInfo: NotaryInformation, notaryContact: NotaryContact, userId: string): Promise<AxiosResponse<User>> {
+    const address = CommonUtils.convertAddressForAuth(notaryInfo?.address)
```
Comment (saravanpa-aot) [inline auth-web/src/services/user.services.ts:94]: do u need ? here ..notaryInfo.address is not enough?

File: auth-web/src/schemas/address-schema.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { maxLength, required } from 'vuelidate/lib/validators'
+
+// The Address schema containing Vuelidate rules.
+// NB: This should match the subject JSON schema.
```
Comment (saravanpa-aot) [inline auth-web/src/schemas/address-schema.ts:4]: what is subject JSON schema??

File: auth-web/src/routes/index.ts [hunk 4]
```diff
@@ -112,12 +112,17 @@ router.beforeEach((to, from, next) => {
...
         return next({ path: `/${Pages.PENDING_APPROVAL}/${currentAccountSettings?.label}` }) // TODO put the account name back once its avaialable ;may be needs a fix in sbc-common
-      } else if (currentAccountSettings && currentMembership.membershipStatus === MembershipStatus.Pending) {
+      } else if (currentAccountSettings && currentMembership?.membershipStatus === MembershipStatus.Pending) {
         console.log('[Navigation Guard] Redirecting user to PENDING_APPROVAL since users membership status is pending')
         return next({ path: `/${Pages.PENDING_APPROVAL}/${currentAccountSettings?.label}` })
-      } else if (!currentOrganization || currentMembership.membershipStatus !== MembershipStatus.Active) {
+      } else if (!currentOrganization || currentMembership?.membershipStatus !== MembershipStatus.Active) {
         console.log('[Navigation Guard] Redirecting user to Create Account since users nerither has account nor an active status')
-        return next({ path: `/${Pages.CREATE_ACCOUNT}` })
+        switch (currentUser?.loginSource) {
```
Comment (saravanpa-aot) [inline auth-web/src/routes/index.ts:120]: are these changes related to canada post?? no note about this change in PR description

File: auth-web/src/models/address.ts [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
+  country?: string
+  street?: string
+  streetAdditional?: string
+  postalCode?: string
+  region?: string
+  deliveryInstructions?: string
+  key?: string
+}
 
+export interface IAddress {
```
Comment (saravanpa-aot) [inline auth-web/src/models/address.ts:12]: just curious , whats the I in IAddress ... may be find a better name...not super picky on this..if possible

File: auth-web/src/components/auth/NonBcscAccounts/NotaryInformationForm.vue [hunk 6]
```diff
@@ -50,14 +55,14 @@ export default class NotaryInformationForm extends Vue {
     notaryName: [v => !!v || 'Name of Notary is required']
   }
 
-  private updateAddress (address: Address) {
-    this.notaryInfo.address = address
+  private updateNotaryAddress (val) {
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/NonBcscAccounts/NotaryInformationForm.vue:58]: is it possible to type val? i guess this is of type address rite?

File: auth-web/src/components/auth/AccountInfo.vue [hunk 7]
```diff
@@ -93,22 +93,14 @@
...
-                </BaseAddress>
-                <div v-if="isAddressViewable">
-                  <div class="value value__title" aria-labelledby="mailingAddress" v-if="currentOrgAddress" >
-                    <div>{{ currentOrgAddress.street }}</div>
-                    <div v-if="currentOrgAddress.streetAdditional">{{ currentOrgAddress.streetAdditional }}</div>
-                    <div>{{ currentOrgAddress.city }}, {{ currentOrgAddress.region }}  {{ currentOrgAddress.postalCode }}</div>
-                    <div>{{ currentOrgAddress.country}}</div>
-                  </div>
-                </div>
+                <base-address
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/AccountInfo.vue:96]: nice ..looks clean

File: auth-web/src/components/auth/AccountInfo.vue [hunk 8]
```diff
@@ -233,12 +230,23 @@ export default class AccountInfoEdit extends Mixins(AccountChangeMixin) {
...
+    }
+  }
+
   private async setup () {
     const accountSettings = this.getAccountFromSession()
     this.orgName = this.currentOrganization?.name || ''
     if (this.isPremiumAccount) {
       await this.syncPaymentSettings(accountSettings.id)
       await this.syncAddress()
+      if (this.currentOrgAddress) {
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/AccountInfo.vue:247]: putting it loud , is there a way to keep one format everywhere and just convert in the service layer when we send to api? 
everywhere else , it can be the BaseAddress format.. and when we send it /receive from api , we do the conversion.. not sure if its practical ,but it will make the code easier

File: auth-web/public/index.html [hunk 9]
```diff
@@ -16,4 +18,8 @@
...
+    // The key for your account.
+    window.addressCompleteKey = 'RH99-RN99-GB72-FW86'
```
[the remark this answers is not in the document]
Comment (nitheesh-aot) [reply auth-web/public/index.html:23]: Yes, will add that in config.json

File: auth-web/src/routes/index.ts (hunk 4 above)
Comment (nitheesh-aot) [reply auth-web/src/routes/index.ts:120]: no, this was not related to canada post, I will add the changes in pr desc

File: auth-web/src/components/auth/AccountInfo.vue (hunk 8 above)
Comment (nitheesh-aot) [reply auth-web/src/components/auth/AccountInfo.vue:247]: the currentOrgAddress is used in multiple places, thats why I converted in the component itself. 
should we change those?

File: auth-web/src/schemas/address-schema.ts (hunk 3 above)
Comment (nitheesh-aot) [reply auth-web/src/schemas/address-schema.ts:4]: comes up while copying from entities code, will remove that

File: auth-web/src/components/auth/AccountInfo.vue (hunk 8 above)
Comment (saravanpa-aot) [reply auth-web/src/components/auth/AccountInfo.vue:247]: hm.. what do you think ? which will be better  any impacts if we change it

File: auth-web/src/schemas/address-schema.ts (hunk 3 above)
Comment (saravanpa-aot) [reply auth-web/src/schemas/address-schema.ts:4]: thanks

**Comment by nitheesh-aot** [comment]
Added a new base address component with address conversion done inside the component itself
